### ISSUE TITLE
debug/di/breakpoint.cpp generates warning C4800.

### ISSUE THREAD

**Issue by sandreenko** [dotnet/coreclr#16289]
The warning is 'BOOL': forcing value to bool 'true' or 'false' (performance warning).
Because all warnings are threated as errors it fails build.
[Example](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_checked_windows_nt_innerloop_flow_prtest/143/console).
Started to hit after #15819.

The full error report:
```
d:\j\workspace\arm64_cross_c---f1ec3c8e\src\debug\di\breakpoint.cpp(210): error C2220: warning treated as error - no 'object' file generated [D:\j\workspace\arm64_cross_c---f1ec3c8e\bin\obj\Windows_NT.arm64.Checked\src\debug\di\cordbdi.vcxproj]
d:\j\workspace\arm64_cross_c---f1ec3c8e\src\debug\di\breakpoint.cpp(210): warning C4800: 'BOOL': forcing value to bool 'true' or 'false' (performance warning) [D:\j\workspace\arm64_cross_c---f1ec3c8e\bin\obj\Windows_NT.arm64.Checked\src\debug\di\cordbdi.vcxproj]
```

**Comment by sandreenko** [comment]
@davmason could you please fix this? I can trigger arm jobs on your PR if you are not in @dotnet/arm64-contrib .

**Comment by davmason** [comment]
@sandreenko I'm happy to fix any issues, but I have to admit I'm a little confused by this. There are tons of places that we check a BOOL in an if statement in the source, do you have any idea what is different about this one?

**Comment by noahfalk** [comment]
I'd be curious from a slightly different angle... these changes were built in numerous configurations including ARM64 when they went through PR earlier and the PR did not fail. Certainly lets fix it, but @sandreenko can you clarify exactly what your repro build process is? If there are simple things we can do to make sure dev and PR builds are aligned then its less hassle for everyone.

FWIW the rolling perf build also tripped up in the same way and that should be a pretty reproducible workflow: https://ci2.dot.net/job/dotnet_coreclr/job/perf/job/master/job/perf_scenarios_Windows_NT_x64_full_opt_ryujit/1186/console
@jorive

**Comment by sandreenko** [comment]
> @sandreenko I'm happy to fix any issues, but I have to admit I'm a little confused by this. There are tons of places that we check a BOOL in an if statement in the source, do you have any idea what is different about this one?

The bad statement is this:
`pEvent->BreakpointData.isIL = m_offsetIsIl; `

where you convert BOOL(int) to bool.

**Comment by sandreenko** [comment]
In other parts of the code we are using (and it was used there before):
`pEvent->BreakpointData.isIL = m_offsetIsIl? true : false;`

**Comment by sandreenko** [comment]
> I'd be curious from a slightly different angle... these changes were built in numerous configurations including ARM64 when they went through PR earlier and the PR did not fail

@noahfalk I do not see windows arm64 job in the PR #15819.

**Comment by davmason** [comment]
Oops, I was looking at the code before my change, where line 210 is an if statement. Thanks for point out my goof. I'll open a PR soon

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by davmason** [comment]
Fixed in #16291

**Comment by noahfalk** [comment]
> @noahfalk I do not see windows arm64 job in the PR #15819.

Agreed, I was looking at the **_ubuntu_** arm64 leg
I hadn't chased through your link earlier so I wasn't aware you were referring specifically to a **_windows_** arm64 build when you mentioned arm64.

Are you aware of any reason that we'd run windows arm64 after checkin is complete but not run it during the PR itself? Hardware limitations?

Thanks!

**Comment by BruceForstall** [comment]
We only run PR testing on Windows arm64 machines (including ARM and ARM64 testing) for people on the `arm64Users` whitelist in netci.groovy, due to security issues. Ideally we could do arm/arm64 builds (but not testing) on all PRs, since the arm64 tools are now public, but haven't done the work to make that happen.

**Comment by noahfalk** [comment]
@BruceForstall - ok, thanks for the info : )

**Comment by jkotas** [comment]
@davmason Could you please resubmit the fix for this? It was reverted together with the other change to fix the build break.

**Comment by davmason** [comment]
@jkotas @sandreenko I included the fix in #16303, and I see it in master at https://github.com/dotnet/coreclr/blob/master/src/debug/di/breakpoint.cpp#L210

I looked through the arm64 windows build job, and the last 4 builds are passing. The ones that failed last night don't seem related to my change.

If there's something still broken can you point it out to me?

**Comment by sandreenko** [comment]
If arm64 job is green then good, the issue is closed.

Thank you.
